library/oid.c

```diff
@@ -623,6 +623,59 @@ FN_OID_GET_OID_BY_ATTR1(mbedtls_oid_get_oid_by_ec_grp,
                         oid_ecp_grp,
                         mbedtls_ecp_group_id,
                         grp_id)
+
+/*
+ * For Elliptic Curve algorithms that are directly
+ * encoded as a top level type (RFC 8410)
+ */
+typedef struct {
+    mbedtls_oid_descriptor_t    descriptor;
+    mbedtls_ecp_group_id        grp_id;
+} oid_ecp_grp_algid_t;
```

Comment:
Sorry if I'm missing something obvious, but this seems to be the exact same type as the existing `oid_ecp_grp_t`, so don't understand why we'd need a new type, new table and new set of getter functions, instead of just adding 4 entries to the existing table `oid_ecp_grp` and its getter functions.

If this is about distinguishing which curves use RFC 8410 encoding and which use RFC 5480 encoding, I'm not sure this is the right way to go about it, as I think it doesn't make it really apparent. If that's the goal, then I think we should do it in a more explicit way: look up all OIDs in the same table, and then to distinguish by curve type, check `grp_id` either directly as you're already doing in some places, or using `mbedtls_ecp_get_type()`.

Reply:
`oid_ecp_grp_algid_t oid_ecp_grp_algid[]` lists OIDs for elliptic curves that can be used an RFC5912 `AlgorithmIdentifier`, i.e. at the same level as `id-ecPublicKey`.

`oid_ecp_grp_t oid_ecp_grp[]` lists OIDs for elliptic curves that can be used as an RFC5480 `namedCurve`.

Reply:
Ok, thanks for clarifying, and sorry for missing it in the first place: the comment does say "that are directly encoded as a top level type".